internal/mithril-persistence/src/database/query/block_range_root/delete_block_range_root.rs

```diff
@@ -0,0 +1,139 @@
+use anyhow::Context;
+use sqlite::Value;
+
+use mithril_common::entities::{BlockNumber, BlockRange};
+use mithril_common::StdResult;
+
+use crate::database::record::BlockRangeRootRecord;
+use crate::sqlite::{Query, SourceAlias, SqLiteEntity, WhereCondition};
+
+/// Query to delete old [BlockRangeRootRecord] from the sqlite database
+pub struct DeleteBlockRangeRootQuery {
+    condition: WhereCondition,
+}
+
+impl Query for DeleteBlockRangeRootQuery {
+    type Entity = BlockRangeRootRecord;
+
+    fn filters(&self) -> WhereCondition {
+        self.condition.clone()
+    }
+
+    fn get_definition(&self, condition: &str) -> String {
+        // it is important to alias the fields with the same name as the table
+        // since the table cannot be aliased in a RETURNING statement in SQLite.
+        let aliases = SourceAlias::new(&[("{:block_range_root:}", "block_range_root")]);
+        let projection = Self::Entity::get_projection().expand(aliases);
+
+        format!("delete from block_range_root where {condition} returning {projection}")
+    }
+}
+
+impl DeleteBlockRangeRootQuery {
+    pub fn include_or_greater_block_number_threshold(
```

Comment:
The name of this function is hard to understand:
```suggestion
    pub fn above_block_number_threshold(
```